Add new options and correct alignment on top panel

miyaokamarina opened the pull request:
-   Align active lines to top when panel is on top edge
-   Option to hide progress margins
-   Option to make active line full-width when progress is active

![Screenshot_20200824_130043](https://user-images.githubusercontent.com/37388187/91031885-db814e00-e609-11ea-9af3-95575ffd2528.png)

miyaokamarina committed:
Add new options and correct alignment on top panel

• Align active lines to top, when panel is on top edge
• Option to hide progress margins
• Option to make active line full-width when prgress is active

psifidotos commented:
Give me some feedback please,

1. why do we need an option to enable/disable the 1px. margin for progress, can you provide some screenshots to understand the benefit? I am asking because if that 1px. margin is not really needed we can remove it totally without providing any option

2. can you provide some screenshots for comparison between active full width when progress is present and without it in order to understand with this is needed?

miyaokamarina commented:
1. Why not. I made it optional because IDK if there was some objectives behind adding 1px margins.
   Switched to Win10 to make sure how its indicators actually look, and I think these margins may be removed completely.
2. By default, the progress bar takes full width of the task button, while its line is sized as usual — full width for active task, reduced for inactive. It looks kind of inaccurate for me because of inconsistent sizing of elements.

![Screenshot_20200824_163634](https://user-images.githubusercontent.com/37388187/91051136-5b1e1580-e628-11ea-832d-b35c32d2575f.png)
![Screenshot_20200824_163701](https://user-images.githubusercontent.com/37388187/91051142-5ce7d900-e628-11ea-841d-afb5ada8b708.png)

---

Switched to Windows to make sure how its indicators actually look. I think, I may add few more tweaks to make these indicators visually compatible:

1. Shrink the progress bar of an inactive task to match its line.
2. Tweak the color of a line with active progress.
3. Follow-on: tweak stacked windows display: Windows doesn’t make distinction between buttons with two or more stacked windows.
4. Follow-on: support panels on left/right edges.

![Screenshot 2020-08-24 132823](https://user-images.githubusercontent.com/37388187/91051466-d7185d80-e628-11ea-8c6c-7488699cfbfd.png)
![Screenshot 2020-08-24 132904](https://user-images.githubusercontent.com/37388187/91051509-e5ff1000-e628-11ea-8f83-6676288dd5a3.png)

psifidotos commented:
> Switched to Win10 to make sure how its indicators actually look, and I think these margins may be removed completely.

no prob if that makes them closer to original W10

> It looks kind of inaccurate for me because of inconsistent sizing of elements.

from my understanding the progress bar either active or inactive it should ALWAYS follow the current line shown size. In that case maybe that should be a specific MR to fix this and at the same remove the previous mentioned margins if they not really needed. In that case the option the make the line bigger when there is progress box is not really needed. Progress box length should follow always the current shown line length.

> Tweak the color of a line with active progress.

I dont like any color specifications from user side so I wont accept it... If you can find a better colors combinations from plasma theme palette to use, I dont have any problem.

>  Windows doesn’t make distinction between buttons with two or more stacked windows.

what do you mean? When there are two or more windows the line breaks to smaller line and slash next to it.

> Follow-on: support panels on left/right edges.

 of couse

miyaokamarina commented:
> I dont like any color specifications from user side so I wont accept it... If you can find a better colors combinations from plasma theme palette to use, I dont have any problem.

I mean, Windows displays a line of the task with progress in a different color.

> When there are two or more windows the line breaks to smaller line and slash next to it.

Here is how a task with three windows looks with these indicators. Well… I think it looks better than in Windows, so I won’t tweak it.

![Screenshot_20200824_171401](https://user-images.githubusercontent.com/37388187/91055361-39735d00-e62d-11ea-869c-700274391a55.png)

BTW, why are most sizes in Latte options specified in percentages instead of pixels?

psifidotos commented:
> I mean, Windows displays a line of the task with progress in a different color.

no problem to change it then as long as the new progress line color is chosen from the plasma theme color palette

> Here is how a task with three windows looks with these indicators.

oh you meant when the task is the active one... the three rectangles approach is based on the Win7 task style, I like it much better in that specific case

> BTW, why are most sizes in Latte options specified in percentages instead of pixels?

Latte provides auto-adjusting size in two important cases:
- when the user does not specify an item size in pixels e.g. 64px but based on the current screen height, e.g. 5%
- when the contents of the dock exceed the screen length and they must be shrunk

so the pixel values would break visually many parts  when the CurrentItemSize is auto-adjusted...

All percentage values are based on the CurrentItemSize used from Latte... So everything is autoadjusted that way.
